src/coffea/analysis_tools.py

```diff
@@ -409,6 +414,507 @@ def variations(self):
+                    arr = dask_awkward.flatten(var[mask])
+                    h.fill(arr, dask_awkward.full_like(arr, i, dtype=int))
+                hists.append(h)
+
+        return hists, labels
+
+
+class Cutflow:
+    """Object to be returned by PackedSelection.cutflow()"""
+
+    def __init__(
+        self, names, nevonecut, nevcutflow, masksonecut, maskscutflow, delayed_mode
+    ):
+        self._names = names
+        self._nevonecut = nevonecut
+        self._nevcutflow = nevcutflow
+        self._masksonecut = masksonecut
+        self._maskscutflow = maskscutflow
+        self._delayed_mode = delayed_mode
+
+    def __repr__(self):
+        return f"Cutflow(selections={self._names})"
+
+    def result(self):
+        """Returns the results of the cutflow as a namedtuple
+
+        Returns
+        -------
+            result : CutflowResult
+                A namedtuple with the following attributes:
+
+                nevonecut : list of integers or dask_awkward.lib.core.Scalar objects
+                    The number of events that survive each cut alone as a list of integers or delayed integers
+                nevcutflow : list of integers or dask_awkward.lib.core.Scalar objects
+                    The number of events that survive the cumulative cutflow as a list of integers or delayed integers
+                masksonecut : list of boolean numpy.ndarray or dask_awkward.lib.core.Array objects
+                    The boolean mask vectors of which events pass each cut alone as a list of materialized or delayed boolean arrays
+                maskscutflow : list of boolean numpy.ndarray or dask_awkward.lib.core.Array objects
+                    The boolean mask vectors of which events pass the cumulative cutflow a list of materialized or delayed boolean arrays
+        """
+        CutflowResult = namedtuple(
+            "CutflowResult",
+            ["labels", "nevonecut", "nevcutflow", "masksonecut", "maskscutflow"],
+        )
+        labels = ["initial"] + list(self._names)
+        return CutflowResult(
+            labels,
+            self._nevonecut,
+            self._nevcutflow,
+            self._masksonecut,
+            self._maskscutflow,
+        )
+
+    def to_npz(self, file, compressed=False):
+        """Saves the results of the cutflow to a .npz file
+
+        Parameters
+        ----------
+            file : str or file
+                Either the filename (string) or an open file (file-like object)
+                where the data will be saved. If file is a string or a Path, the
+                ``.npz`` extension will be appended to the filename if it is not
+                already there.
+            compressed : bool, optional
+                If True, the data will be compressed in the ``.npz`` file.
+                Default is False.
+        """
+        labels, nevonecut, nevcutflow, masksonecut, maskscutflow = self.result()
+        if self._delayed_mode:
```

Comment:
do we want this to always, explicitly start a computation?

For example `dask_awkward.to_parquet` provides the possibility of doing its computation later, depending on the opinion of the user.